### PR TITLE
chore(ci): ignore generated code in codeql result

### PR DESCRIPTION
**Pull request by menski**
## Description

After the CodeQL analysis filter out all results related to code in a `generated-sources` or `generated-test-sources` folder.
    
Solution based on https://josh-ops.com/posts/github-codeql-ignore-files/

### CODE REVIEW

**Review by lenaschoenburg — APPROVED** [review]
Thanks! :heart_eyes:

File: .github/workflows/code-quality.yml [hunk 1]
```diff
@@ -37,6 +37,21 @@ jobs:
         run: mvn -B -T1C -DskipTests -DskipChecks install
       - name: Perform CodeQL Analysis
         uses: github/codeql-action/analyze@v2
+        with:
+          upload: False
+          output: sarif-results
+      - name: filter-sarif
```
[the remark this answers is not in the document]
Comment (menski) [reply .github/workflows/code-quality.yml]: Good idea :see_no_evil:

**Comment by menski** [comment]
bors retry

**Pull request merged**